rust/core-lib/src/view.rs

```diff
@@ -665,9 +665,10 @@ impl View {
     /// next occurrence before (`true`) or after (`false`) the last cursor is selected. `wrapped`
     /// indicates a search for the next occurrence past the end of the file. `stop_on_found`
     /// determines whether the search should stop at the first found occurrence (does only apply
-    /// to forward search (i.e. reverse = false).
+    /// to forward searc, i.e. reverse = false). If `allow_same` is set to `true` the current
+    /// selection is considered a valid next occurrence.
     pub fn select_next_occurrence(&mut self, text: &Rope, reverse: bool, wrapped: bool,
-                                  stop_on_found: bool) -> Option<usize>
+                                  stop_on_found: bool, allow_same: bool) -> Option<usize>
```

Comment:
I am not particularly proud that I had to add another boolean option here 😕 and I am wondering whether using flags for `reverse`, `wrapped`, `stop_on_found` and `allow_same` would be a better approach?